WcaOnRails/app/controllers/api/v0/records_controller.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+class Api::V0::RecordsController < Api::V0::ApiController
+  def regionCondition(countrySource)
+    if params[:regionId]
+      region = params[:regionId]
+
+      if /^(world)?$/ =~ region
+        ''
+      elsif /^_/ =~ region
+        return "AND continentId = \'#{region}\'"
+      else
+        if countrySource
+          countrySource += '.'
+        end
+
+        return "AND #{countrySource}countryId = \'#{region}\'"
+      end
+    end
+
+    ''
```

Comment:
I'd start the method with `result = ''`, end it with `result`, replace any `return X` with `result = X`, and trim out a few lines.